### DISCUSSION TITLE
Questions on Pausability and `selfdestruct` Functions

### DISCUSSION THREAD

**Discussion by 0xD006F00D** [pcaversaccio/xdeployer#34] [Q&A] [answered]
Hi, two quick questions:

What was the reasoning for making the Create2Deployer contract implement selfdestruct and the pausable interface?

What other issues might arise in the future using somebody else's Create2Deployer contract? 
One disadvantage I can see is that if the projects is no longer maintained but another EVM chain comes along there is no way to redeploy the deployer to the new chain (with its current address). 
Anything else to be aware of?

Thanks

**Comment by pcaversaccio** [comment, marked as answer]
Hi,

maybe to give some context for other people, the discussion relates to the underlying [`Create2Deployer`](https://github.com/pcaversaccio/create2deployer/blob/main/contracts/Create2Deployer.sol) contract. So the reasoning at the beginning was that in case something does not work smoothly during the `CREATE2` deployment, I could disable the `deploy` (and `deployERC1820Implementer`) functionality in order to debug without having people wasting money on potential flawed deployments. However, after so many months it seems everything works as intended - so this is more of a legacy, debugging feature I kept for all contracts in order to be the same on all chains. The `selfdestruct` possibility was/is there in order for me to replay the `CREATE2` feature of deploying, destructing and redeploying feature with the same address. Also, I kept it as an _emergency backup_ parallel to the `pausable` feature.

Everything is **open-source**; anyone can redeploy the `CREATE2` deployer across all chains and fork the `xdeployer` plugin and adjust accordingly. So I don't see any risk tbh. The goal of `xdeployer` is to provide a simple but extensible plugin for `CREATE2` deployments. The design goal is to keep it as simple as possible since already enough complexity lurks in the back. Anyone is free to write Hardhat tasks around it - it's really just focused on doing one thing properly: **deploying one contract with the same contract address on all selected EVM chains.**

**Comment by pcaversaccio** [comment]
**Note**: As of 8 December 2023, all non-deprecated [`Create2Deployer`](https://github.com/pcaversaccio/create2deployer/blob/main/contracts/Create2Deployer.sol) deployments are permissionless as I have renounced the ownership accordingly.
